tests/acceptance/29_audit_mode/audit_safe_functions.cf

```diff
@@ -0,0 +1,16 @@
+bundle agent __main__
+{
+  vars:
+    "cf_agent" string => ifelse(isvariable("sys.cf_agent"), "$(sys.cf_agent)", "/var/cfengine/bin/cf-agent");
+#    "result" string => execresult("$(cf_agent) --audit -Kf $(this.promise_filename).sub", useshell);
+    "result" string => execresult("$(cf_agent) -Kf $(this.promise_filename).sub", useshell);
+
+  reports:
+    "result is $(result)";
+    "$(this.promise_filename) Pass"
+      if => regcmp("R:.* Pass", "$(result)");
+    "$(this.promise_filename) FAIL"
```

Comment:
I think that you can use one of the dcs bundles for this.

Perhaps

- bundle agent =dcs_check_regcmp(regex, thestring, test, expected_mismatch)= :: Pass/FAIL
     based on =expected_mismatch= (=yes=/=no=) when the regular expression
     =regex= matches the string =thestring=.